Drug Dealer Simulator 2525

Raikyr0 opened the pull request:
![DrugDealerSimPR](https://github.com/user-attachments/assets/c6689618-a305-453b-8edc-1fdf73e0a569)


# Description

Adds a sell price for most narcotics with street use. Values are based on chemical complexity.

# Changelog

:cl:
- add: Added sell price for illegal narcotics.

Raikyr0 committed:
Update narcotics.yml

Raikyr0 committed:
Update narcotics.yml

Raikyr0 committed:
Update narcotics.yml

Raikyr0 committed:
Update medicine.yml

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update narcotics.yml

Raikyr0 committed:
Update medicine.yml

Raikyr0 committed:
Update narcotics.yml

Raikyr0 committed:
Update natural_sauce.yml

Raikyr0 committed:
Update narcotics.yml

M3739 commented on Resources/Prototypes/_Floof/Reagents/natural_sauce.yml:
```diff
@@ -6,6 +6,7 @@
   physicalDesc: reagent-physical-desc-sticky
   flavor: salty
   color: "#ffffff"
+  pricePerUnit: 20 # Floof - This one is just here because it'll be funny to sell cum
```
Unfortunately, this leads to **arbitrage** with the lewd toys crate.
 **Buy $1200 -> Sell $4900**
<img width="342" height="100" alt="image" src="https://github.com/user-attachments/assets/b67d8ceb-c78f-4a8f-8dd6-ee1258f7f4f9"/>

**Remove the `pricePerUnit`, or make it so cheap where arbitrage is no longer possible.**

M3739 reviewed (changes requested):
This... Will need some discussion. Adding chemical sell prices such as these will make the chemists much more involved in the economy of the station. 

However, **one major issue needs to be addressed.**

Raikyr0 committed:
Update natural_sauce.yml

M3739 commented on Resources/Prototypes/_Floof/Reagents/natural_sauce.yml:
```diff
@@ -6,6 +6,7 @@
   physicalDesc: reagent-physical-desc-sticky
   flavor: salty
   color: "#ffffff"
+  pricePerUnit: 5 # Floof - This one is just here because it'll be funny to sell cum
```
Still viable for arbitrage.
Buy $1200 -> Sell $1900
<img width="335" height="108" alt="image" src="https://github.com/user-attachments/assets/4f1f7ce2-a869-4f9b-9838-b258b0f61abd" />

Raikyr0 committed:
Update natural_sauce.yml

M3739 reviewed (commented):
Arbitrage has been addressed. Now we await the debates.

Raikyr0 committed:
Merge branch 'Floof-Station:master' into DrugDealerSimulator2525

Raikyr0 committed:
Merge branch 'Floof-Station:master' into DrugDealerSimulator2525

Memeji commented:
Adding more items to the economy sounds fun to me. Might be my inner tycoon talking though.

Chokovit commented:
Hello, I request this be unmarked as stale. I highly support this PR!

M3739 commented:
Note to self, perform reassessment of this PR sometime.

Memeji reviewed (approved):
I'm going to give this my approval as I like the idea. A lot of the balance changes will need to be discovered as it gets used, in order to see what is too much/what is too little.

We'll reach that road when we get there. For now though, it's a nice addition imo.

Raikyr0 committed:
Added Value to some Poisons, because there's probably a market for it

Raikyr0 committed:
Update narcotics.yml

Raikyr0 commented:
Added some value to Vestine so Sec doesn't feel the need to cook something with it to sell for more.
Also added value to things that use Vestine, so if sec collects it they can sell it off.

Chokovit commented:
We have 1 maintainer who approves can we get second approval?

Mnemotechnician reviewed (changes requested):
30K for a bottle of nocturine is insane. Iirc thieves get 3 of those in a single kidnapping kit - that's 90k right off the start, and knowing the culture of floof, some people WILL give those bottles to cargo under a "don't snitch" promise.

I'd say those should give at MOST 10k/50 units. Things that can be easily synthesized in a chemistry lab should be worth even less because it's just an insane arbitrage on chem crates that people can abuse.

Raikyr0 commented:
> 30K for a bottle of nocturine is insane. Iirc thieves get 3 of those in a single kidnapping kit - that's 90k right off the start, and knowing the culture of floof, some people WILL give those bottles to cargo under a "don't snitch" promise.
> 
> I'd say those should give at MOST 10k/50 units. Things that can be easily synthesized in a chemistry lab should be worth even less because it's just an insane arbitrage on chem crates that people can abuse.

I was unaware that the kidnapping kit for thieves (they're so rare and do zero kidnapping) on our server had been buffed from 1 to 3 nocturine bottles, the price is mostly in line with the TC cost of a bribery bundle for syndies.

As for chem crate arbitrage, it is intended as a way for chem to make money, and the tedious act of processing the chemical creates value. Much like how a sweatshop turns a dollar of materials into thousands of dollars in brand shoes or smart phones.
Labor generates value.

Chokovit commented:
I'm not very concerned with powergaming or meta strategy with this. Security exists to balance those things. Making illegal chems SHOULD cost what Raikyr made them worth because the illicit nature of making them is what causes the increased risk vs reward.

However, I do also agree that maybe Nocturine could be $10k for a full bottle? I didn't know thieves could start with 3 of them.

Raikyr0 commented:
> I'm not very concerned with powergaming or meta strategy with this. Security exists to balance those things. Making illegal chems SHOULD cost what Raikyr made them worth because the illicit nature of making them is what causes the increased risk vs reward.
> 
> However, I do also agree that maybe Nocturine could be $10k for a full bottle? I didn't know thieves could start with 3 of them.

For now, I've decided against changing it, more of a "the changelog doesn't mention which chemicals have a sell value" so they'll have to figure it out, and nocturine isn't very obvious unless someone looked through the code. I'd like to run it through live first to gauge how much tweaking needs to be done, and if a thief does in fact sell their nocturine more then once. Thieves are also pretty rare so.

Mnemotechnician commented:
> Security exists to balance those things.

In my 3000 hours of SS14 playtime, about 50% of which was on Floofstation, I have only witnessed about a dozen cases of security actually caring about drug production and distribution. And usually it was due to oracle spitting out the zombie drug, not due to chem cooking nocturine. When it comes to game-breaking cargo arbitrage, we cannot simply rely on players to balance it out.

Moreover, the uplink is not the only source of nocturine and other chemicals. Artifacts can spit it out. Reagent slimes (sources from your local critted chemical anomaly or random vent events) can easily give you ~60-100 units of this stuff per catch - and that's without even abusing blood bags. Oneirophages (currently disabled) have it in their blood. Chem anomalies can inject it into beakers. It's in the hydroponics chem content mutation pool. It can be found in maint pills and shrooms.

You get the idea.

Chokovit commented:
> > Security exists to balance those things.
> 
> In my 3000 hours of SS14 playtime, about 50% of which was on Floofstation, I have only witnessed about a dozen cases of security actually caring about drug production and distribution. And usually it was due to oracle spitting out the zombie drug, not due to chem cooking nocturine. When it comes to game-breaking cargo arbitrage, we cannot simply rely on players to balance it out.
> 
> Moreover, the uplink is not the only source of nocturine and other chemicals. Artifacts can spit it out. Reagent slimes (sources from your local critted chemical anomaly or random vent events) can easily give you ~60-100 units of this stuff per catch - and that's without even abusing blood bags. Oneirophages (currently disabled) have it in their blood. Chem anomalies can inject it into beakers. It's in the hydroponics chem content mutation pool. It can be found in maint pills and shrooms.
> 
> You get the idea.

Hmm, that makes a little more sense.

 Buy at the same time I DO want there to feel like there is a risk vs reward for this.

Raikyr0 commented:
> > Security exists to balance those things.
> 
> In my 3000 hours of SS14 playtime, about 50% of which was on Floofstation, I have only witnessed about a dozen cases of security actually caring about drug production and distribution. And usually it was due to oracle spitting out the zombie drug, not due to chem cooking nocturine. When it comes to game-breaking cargo arbitrage, we cannot simply rely on players to balance it out.
> 
> Moreover, the uplink is not the only source of nocturine and other chemicals. Artifacts can spit it out. Reagent slimes (sources from your local critted chemical anomaly or random vent events) can easily give you ~60-100 units of this stuff per catch - and that's without even abusing blood bags. Oneirophages (currently disabled) have it in their blood. Chem anomalies can inject it into beakers. It's in the hydroponics chem content mutation pool. It can be found in maint pills and shrooms.
> 
> You get the idea.

You make a valid argument that there are alternative sources of drugs, but the main balancing is around Walter Whiting it. 
All other sources are based on RNG, very much unreliable. Unless someone goes out of their way as a non-antag to crit a chemical anomaly just to make money which an admin totally should backhand them for that.
